sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.java

```diff
@@ -0,0 +1,270 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Check that forbidden method is not used. We can forbid a method by method name and number of
+ * arguments.
+ * This can be used to enforce things like:
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() and Assert.assertFalse() have a 1 arg variant that does not provide a
+ * helpful message on failure.
```

Comment:
by this examples it looks like you can reliably catch such methods , but this Check can not do this reliably and produce a lot of false positives as it catch all "exist" method, even unrelated.

1) Nuance of implementation should be described in javadoc as ATTENTION note.
2) "exit" can not be used as default, as it catch too much false positives. Lets try "assert(True|False)" it it less common.

Reply:
@romani Can you point me to these false positives ?

Reply:
@raghavgautam Here is one I found:
https://raghavgautam.github.io/infinispan/index.html#A6
https://raghavgautam.github.io/infinispan/xref/Users/raghav/play/checkstyle-contribution/checkstyle-tester/repositories/infinispan/core/src/test/java/org/infinispan/distribution/rehash/PessimisticStateTransferLocksTest.java.html#L155
> sequencer.exit("tx:perform_op");

Reply:
I have generated two more reports and uploaded them.
https://raghavgautam.github.io/halt/
https://raghavgautam.github.io/assert/
https://raghavgautam.github.io/exit/

I am curious if there any other code base that has defined exit method in addition to infinispan?

Reply:
"exit" is too general name.
we almost randomly selected such projects, and one of them do have it, so chance if pretty big.
"assertTrue" is bigger name unlikely to be reused for smth else.

So we just change default to be correct by default.
If user take responsibility on possible false-positives and understand that some undesirable match could happen he could provide any name , even "run".
But in documentation we have to explain him how Check works, all nuances of his validation.

Reply:
Will make the change.